javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -488,14 +491,24 @@ protected void generatePagingAsync(ClientMethod clientMethod, JavaType typeBlock
         if (clientMethod.getMethodPageDetails().nonNullNextLink()) {
             writeMethod(typeBlock, clientMethod.getMethodVisibility(), clientMethod.getDeclaration(), function -> {
                 addOptionalVariables(function, clientMethod, restAPIMethod.getParameters(), settings);
+                if (settings.isLowLevelClient()) {
+                    function.line("RequestOptions requestOptionsForNextPage = new RequestOptions();");
+                    function.line("requestOptionsForNextPage.setContext(requestOptions != null ? requestOptions.getContext() : Context.NONE);");
+                }
                 function.line("return new PagedFlux<>(");
                 function.indent(() -> {
                     function.line("() -> %s(%s),",
                             clientMethod.getProxyMethod().getPagingAsyncSinglePageMethodName(),
                             clientMethod.getArgumentList());
-                    function.line("nextLink -> %s(%s));",
-                            clientMethod.getMethodPageDetails().getNextMethod().getProxyMethod().getPagingAsyncSinglePageMethodName(),
-                            clientMethod.getMethodPageDetails().getNextMethod().getArgumentListWithoutRequestOptions());
+                    if (settings.isLowLevelClient()) {
+                        function.line("nextLink -> %s(%s));",
+                                clientMethod.getMethodPageDetails().getNextMethod().getProxyMethod().getPagingAsyncSinglePageMethodName(),
+                                clientMethod.getMethodPageDetails().getNextMethod().getArgumentList().replace("requestOptions", "requestOptionsForNextPage"));
+                    } else {
+                        function.line("nextLink -> %s(%s));",
+                                clientMethod.getMethodPageDetails().getNextMethod().getProxyMethod().getPagingAsyncSinglePageMethodName(),
+                                clientMethod.getMethodPageDetails().getNextMethod().getArgumentListWithoutRequestOptions());
```

Comment:
Is this same as `getArgumentList` if not LLC? And probably remove/comment-out `getArgumentListWithoutRequestOptions` method?

Reply:
should be the same for non-llc, I will comment-out `getArgumentListWithoutRequestOptions` and make llc, non-llc consistemt.